src/main/java/frc/robot/subsystems/drivetrain/commands/OverpoweredDrive.java

```diff
@@ -0,0 +1,111 @@
+package frc.robot.subsystems.drivetrain.commands;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.subsystems.drivetrain.SwerveDrive;
+
+import java.util.function.DoubleSupplier;
+
+import static frc.robot.Constants.SwerveDrive.VELOCITY_MULTIPLIER;
+
+/*
+    OverpoweredDrive is an assisted driving command that provides a smoother driving experience.
+    It corrects heading drifting, slows acceleration, and waits for wheels to reach their desired rotations before speeding.
+ */
+public class OverpoweredDrive extends HolonomicDrive {
+    private final PIDController pidController = new PIDController(Constants.SwerveDrive.HEADING_KP, 0, 0) {{
+        enableContinuousInput(-Math.PI, Math.PI);
+        setTolerance(Math.toRadians(Constants.SwerveDrive.ALLOWABLE_HEADING_ERROR));
+    }};
+    private boolean newSetpoint = false;
+    private Rotation2d setpoint;
+    private boolean wait = false;
+    private double current = 0;
```

Comment:
Why don't you use SlewRateLimiter?

Reply:
It is possible and it's not too different. But, I like this better for the fact that it doesn't go above the magnitude. Which slew can when going from high to low. Which makes it feel less responsive.